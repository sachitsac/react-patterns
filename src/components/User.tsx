import { ReactElement } from "react";
import type { User } from "../types/User";

/**
 * This component is used to display the user details
 * Notice there is only ui logic here and not any business logic
 * This is a presentational component
 *
 * @param props
 * @returns ReactElement
 */
export const UserComponent = (props: { user: User }): ReactElement => {
  return (
    <div>
      <h1>{props.user.name}</h1>
      <p>{props.user.email}</p>
    </div>
  );
};
