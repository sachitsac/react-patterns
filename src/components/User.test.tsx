import { render, screen } from "@testing-library/react";
import { UserComponent } from "./User";
import { User } from "../types/User";

const user: User = {
  id: "user-1",
  name: "John Doe",
  email: "john.doe@example.com",
};

const setup = (user: User) => {
  render(<UserComponent user={user} />);
};

describe("UserComponent", () => {
  it("should render user details", () => {
    const component = setup(user);
    expect(screen.getByText(user.name)).toBeInTheDocument();
    expect(screen.getByText(user.email)).toBeInTheDocument();
    expect(component).toMatchSnapshot();
  });
});
