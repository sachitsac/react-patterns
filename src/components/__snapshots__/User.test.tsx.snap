// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserComponent > should render user details 1`] = `undefined`;
